deepchem/models/tensorflow_models/lr.py

```diff
@@ -0,0 +1,231 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Tue Nov 08 14:10:02 2016
+
+@author: Zhenqin Wu
+"""
+import tensorflow as tf
+import numpy as np
+import os
+import time
+
+from deepchem.metrics import from_one_hot
+from deepchem.models.tensorflow_models import TensorflowGraph
+from deepchem.models.tensorflow_models import TensorflowGraphModel
+from deepchem.models.tensorflow_models import model_ops
+from deepchem.utils.save import log
+from deepchem.data import pad_features
+from deepchem.metrics import to_one_hot
+
+def weight_decay(penalty_type, penalty):
+  # due to the different shape of weight(ndims=2) and bias(ndims=1),
+  # will using this version for logreg
+  variables = []
+  # exclude bias variables
+  for v in tf.trainable_variables():
+    if v.get_shape().as_list()[0] > 1:
+      variables.append(v)
+
+  with tf.name_scope('weight_decay'):
+    if penalty_type == 'l1':
+      cost = tf.add_n([tf.reduce_sum(tf.abs(v)) for v in variables])
+    elif penalty_type == 'l2':
+      cost = tf.add_n([tf.nn.l2_loss(v) for v in variables])
+    else:
+      raise NotImplementedError('Unsupported penalty_type %s' % penalty_type)
+    cost *= penalty
+    tf.scalar_summary('Weight Decay Cost', cost)
+  return cost    
+    
+    
+class TensorflowLogisticRegression(TensorflowGraphModel):
```

Comment:
Ah I see. That's an entirely reasonable answer. Would you mind adding a simple test to `models/tests/test_overfit.py` that tests the new tf logistic models?